utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,651 @@
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+		runtime_common::claims::PrevalidateAttests::<Runtime>::new(),
+	)
+}
+
+fn signed_ext_builder_kusama(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> kusama_runtime_exports::SignedExtra {
+	use kusama_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+fn signed_ext_builder_westend(
+	nonce: Index,
+	tip: Balance,
+	era: sp_runtime::generic::Era,
+) -> westend_runtime_exports::SignedExtra {
+	use westend_runtime_exports::Runtime;
+	(
+		frame_system::CheckSpecVersion::<Runtime>::new(),
+		frame_system::CheckTxVersion::<Runtime>::new(),
+		frame_system::CheckGenesis::<Runtime>::new(),
+		frame_system::CheckMortality::<Runtime>::from(era),
+		frame_system::CheckNonce::<Runtime>::from(nonce),
+		frame_system::CheckWeight::<Runtime>::new(),
+		pallet_transaction_payment::ChargeTransactionPayment::<Runtime>::from(tip),
+	)
+}
+
+construct_runtime_prelude!(polkadot, NposCompactSolution16);
+construct_runtime_prelude!(kusama, NposCompactSolution24);
+construct_runtime_prelude!(westend, NposCompactSolution16);
+
+#[macro_export]
+macro_rules! any_runtime {
+	($($code:tt)*) => {
+		unsafe {
+			match $crate::RUNTIME {
+				$crate::AnyRuntime::Polkadot => {
+					use $crate::polkadot_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Kusama => {
+					use $crate::kusama_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Westend => {
+					use $crate::westend_runtime_exports::*;
+					$($code)*
+				}
+			}
+		}
+	}
+}
+
+#[derive(codec::Encode, codec::Decode, Clone, Copy, Debug)]
+#[allow(unused)]
+enum MinerProfile {
+	/// seq-phragmen -> balancing(round) -> reduce
+	WithBalancing(u32),
+	/// seq-phragmen -> reduce
+	JustSeqPhragmen,
+	/// trim the least staked `perbill%` nominators, then seq-phragmen -> reduce
+	TrimmedVoters(sp_runtime::Percent),
+	/// Terminate. There's nothing else that we can do about this. Sorry.
+	Terminate,
+}
+
+#[allow(unused)]
+impl MinerProfile {
+	/// Get the next miner profile to use, should this one fail.
+	fn next(self) -> Self {
+		use sp_runtime::Percent;
+		match self {
+			MinerProfile::WithBalancing(count) => {
+				if count > 0 {
+					MinerProfile::WithBalancing(count.saturating_sub(3))
+				} else {
+					MinerProfile::JustSeqPhragmen
+				}
+			}
+			MinerProfile::JustSeqPhragmen => MinerProfile::TrimmedVoters(Percent::from_percent(90)),
```

Comment:
is 90% also a known hard-coded thing?

and 10 percent below

Reply:
This was just an idea, I removed it already.